From b67bc12cfe374e5139d43edb26a2932a5f0bae00 Mon Sep 17 00:00:00 2001
From: Babu Shanmugam <bschanmu@redhat.com>
Date: Tue, 15 Nov 2016 12:05:49 +0000
Subject: [PATCH 1/2] Input DB listen IP address for the northd

When ovn-northd is started, along with it ovsdb servers for the southbound
and northbound databases are also started. Without additional options for the
northd service, the DB servers will be listenting on 0.0.0.0

This patch creates a systemd environment file that will create the db listen
options for the northd service.

Change-Id: I99b1c3e0fe89b763be81db9b8c1c3b257be42e6e
---
 manifests/northd.pp             | 18 ++++++++++++++++--
 spec/classes/ovn_northd_spec.rb | 14 ++++++++++++++
 2 files changed, 30 insertions(+), 2 deletions(-)

diff --git a/manifests/northd.pp b/manifests/northd.pp
index 55dd6e4..44d0c2b 100644
--- a/manifests/northd.pp
+++ b/manifests/northd.pp
@@ -1,10 +1,24 @@
 # ovn northd
 # == Class: ovn::northd
 #
-# installs ovn package starts the ovn-northd service
+# Installs ovn package starts the ovn-northd service
 #
-class ovn::northd() {
+# [*dbs_listen_ip*]
+#   The IP-Address where OVN DBs should be listening
+#   Defaults to '0.0.0.0'
+#
+class ovn::northd($dbs_listen_ip = '0.0.0.0') {
     include ::ovn::params
+    if $::osfamily == 'RedHat' {
+      file { '/etc/sysconfig/ovn-northd':
+        ensure  => file,
+        mode    => '0600',
+        owner   => 'root',
+        group   => 'root',
+        content => "OVN_NORTHD_OPTS=--db-nb-addr=${dbs_listen_ip} --db-sb-addr=${dbs_listen_ip}",
+        before  => Service['northd']
+      }
+    }
 
     service { 'northd':
         ensure    => true,
diff --git a/spec/classes/ovn_northd_spec.rb b/spec/classes/ovn_northd_spec.rb
index 320377c..50a8f47 100644
--- a/spec/classes/ovn_northd_spec.rb
+++ b/spec/classes/ovn_northd_spec.rb
@@ -2,6 +2,19 @@ require 'spec_helper'
 
 describe 'ovn::northd' do
 
+  shared_examples_for 'systemd env' do
+    it 'creates systemd conf' do
+      is_expected.to contain_file('/etc/sysconfig/ovn-northd').with(
+        :ensure  => 'file',
+        :mode    => '0600',
+        :owner   => 'root',
+        :group   => 'root',
+        :content => "OVN_NORTHD_OPTS=--db-nb-addr=0.0.0.0 --db-sb-addr=0.0.0.0",
+        :before  => 'Service[northd]',
+      )
+    end
+  end
+
   shared_examples_for 'ovn northd' do
     it 'includes params' do
       is_expected.to contain_class('ovn::params')
@@ -56,6 +69,7 @@ describe 'ovn::northd' do
           }
         end
         it_behaves_like 'ovn northd'
+        it_behaves_like 'systemd env'
       end
     end
   end
-- 
2.9.3

